plugins/jobtype/test/azkaban/jobtype/TestHadoopJobUtilsFindApplicationIdFromLog.java

```diff
@@ -0,0 +1,120 @@
+package azkaban.jobtype;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Set;
+
+import org.apache.log4j.Logger;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class TestHadoopJobUtilsFindApplicationIdFromLog {
```

Comment:
@Test
  public void testSparkPreSubmit() throws IOException 

tests this scenario
(the function will return a Set<String> with no elements in it)